src/front-end/src/components/Header/index.tsx

```diff
@@ -33,7 +33,7 @@ const Header = () => {
 
   React.useEffect(() => {
     const encoded_state: string | null = localStorage.getItem("state");
-    if (encoded_state) {
+    if (encoded_state && encoded_state !== "undefined") {
```

Comment:
Why do we need this guard?

Reply:
without it I was seeing this in the console:
```bash
"undefined" is not valid JSON
```